backend/testClient.spec.js

```diff
@@ -0,0 +1,216 @@
+const { createTestClient } = require('apollo-server-testing');
+
+const { typeDefs } = require("./typeDefs");
+const { resolvers } = require("./resolvers");
+
+const { ApolloServer, gql } = require('apollo-server');
+
+const { jwt } = require('jsonwebtoken');
+
+const { v1 } = require('neo4j-driver');
+
+const { makeAugmentedSchema } = require('neo4j-graphql-js');
+const schema = makeAugmentedSchema ({ typeDefs , resolvers });
+
+let token;
+let query;
+let mutate;
+
+describe("ServerTest", () => {
+
+  beforeAll(async () => {
+    const neoDriver = v1.driver(
+      'bolt://localhost:7687',
+      v1.auth.basic('neo4j', 'password')
+    );
+
+    const server = new ApolloServer({
+      schema
+    });
+
+    server.requestOptions = {
+      context() {
+        return {
+          token: token,
+          neoDriver
+        }
+      }
+    };
+
+    query = createTestClient(server).query;
+    mutate = createTestClient(server).mutate;
+  });
+
+  it('test Login', async () => {
+    const res = await mutate({ mutation: LOGIN_USER,
+      variables: { username: "user1", password: "12345"}
+    });
+
+    token = res.data.loginUser.token;
+
+    let res2 = await query({ query: GET_TODOS});
+
+    console.log(res2);
+    expect(res2).toMatchObject({
+      "data": {
+        "todos": [
+          {
+            "title": "SDF - Task 3",
+          },
+          {
+            "title": "SDF - Task 2",
+          }
+        ]
+      }
+    })
+  });
+
+  it('get all todos', async () => {
+    const res = await query({ query: GET_TODOS, variables: {token: token}
+    });
+    
+    expect(res).toMatchObject(
```

Comment:
Use jest's async expectations:
```suggestion
    await expect(query({ query: GET_TODOS, variables: {token: token})).resolves.toMatchObject(/* ... */)
```
and get much better failure explanations.